Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -12,21 +12,6 @@ internal typealias CoreConfiguration = FeaturesConfiguration.Common
 /// Feature-agnostic set of dependencies powering Features storage, upload and event recording.
 internal typealias CoreDependencies = FeaturesCommonDependencies
 
-/// A shim interface for allowing V1 Features generic initialization in `DatadogCore`.
-internal protocol V1FeatureInitializable {
-    /// The configuration specific to this Feature.
-    /// In V2 this will likely become a part of the public interface for the Feature module.
-    associatedtype Configuration
-
-    init(
-        storage: FeatureStorage,
-        upload: FeatureUpload,
-        configuration: Configuration,
-        commonDependencies: FeaturesCommonDependencies,
-        telemetry: Telemetry?
-    )
-}
-
```

Comment:
My issue with this protocol is that it creates a dependency of Features to the `DatadogCore`, we should avoid coupling those two layers.
I'm aware that `FeaturesConfiguration.Common` and `FeaturesCommonDependencies` are also coupling those two, I'm considering moving those configuration to the new `DatadogV1CoreProtocol`, WDYT?

Reply:
Reverted 👍